src/deluge/gui/ui/keyboard/layout.h

```diff
@@ -51,7 +51,9 @@ class KeyboardLayout {
 	virtual void handleVerticalEncoder(int32_t offset) = 0;
 
 	/// Will be called with offset 0 to recalculate bounds on clip changes
-	virtual void handleHorizontalEncoder(int32_t offset, bool shiftEnabled) = 0;
+	virtual void handleHorizontalEncoder(int32_t offset, bool shiftEnabled,
```

Comment:
This is the only change that I'm not sure about since it sort of breaks the metaphor of scrolling the keyboard under your fingers

That said I'm also not sure how useful that is beyond familiarity since you'll usually just have one pad held in this keyboard anyway, and your discord posts about it really speak to how useful this is so I think we should give it a try

Reply:
For documentation, this is what I put in the discord

"The reason I wasn't doing that was with the other keyboards SHIFT + :hturn:  had more to do with the keyboard layout.

If it is SHIFT, and you have to be pressing a chord to change voicings, it pretty much requires sticky SHIFT. It would be near impossible to hold SHIFT  + :hturn:  + :padpress: . I don't know how much that is a draw back.

Then considerations for future next steps,

I want to add some more functionality like changing Chord order, which should be fairly easy to do if handleVerticalEncoder is passed PressedPad presses[kMaxNumKeyboardPadPresses], then you could just drag them like you drag notes in clip view or song view.

The bigger thing is that I would like to maybe add an ability to change notes from one chord to the next inside a family. If you are on a G7, you could move to G9, move to G7b9, etc. This would potentially allow reducing the height of the chord keyboard, maybe. But then I don't know how to do that controls wise. I thought about maybe :vpressturn: + :padpress:  or SHIFT + :vturn:/:hturn:   + :padpress:  could be used for this. It feels like the chord keyboard might just have more functionality than the other keyboards.

If we had horizontal dimmer or white bars (every 4 rows or whatever) for easier navigation, that could be configurable with shift + :hturn:  or we could use SHIFT + :hturn:  to offset the root note row octave wise or allow more rows of the root note for playing a bass line while doing chords. 

Just things I have thought about for other useful features and controls with the chord keyboard for the next round. I didn't want to make this initial PR too big
Also since I have so many future plans for the chord keyboard feature and control wise. I was thinking if it does make it into 1.2, putting it behind a community features menu for now. I don't think it needs to be behind one forever since it doesn't impact basically anything when it isn't in use, but if it is behind that menu it allow it to be more fluid to change until it is finalized"